src/util/cmdline.cpp

```diff
@@ -145,6 +148,42 @@ const char *cmdlinet::getval(const char *option) const
   return value->second.front().c_str();
 }
 
+std::string cmdlinet::expand_path_string(std::string path) const
+{
+  return path;
+}
+
+// Will return empty string if invalid.
+std::string cmdlinet::get_config_file_location() const
+{
+  const auto envloc = std::getenv("ESBMC_CONFIG_FILE");
+  if (envloc)
+  {
+    const std::string config_path =
+      this->expand_path_string(std::string(envloc));
+    if (std::filesystem::exists(config_path))
+    {
+      return config_path;
+    }
+  }
+  else
+  {
+    // Load default config file if it exists.
+#ifdef WIN32
+#  define DEFAULT_CONFIG_PATH "%userprofile%\\esbmc.toml"
+#else
+#  define DEFAULT_CONFIG_PATH "~/.config/esbmc.toml"
+#endif
```

Comment:
better to search for the config file in the current running dir.

Reply:
Tools usually have a list of paths to search for config files in. It could include (paths starting from) the current working directory (at the top), like clang-format, but doesn't have to. Also, system-wide settings in `/etc` are typically used whenever a config file in the user's home directory doesn't override it.

Reply:
Should I add a list of locations to look for a file:

1. Current directory
2. User config folder? `~/.config/esbmc.toml`

Reply:
Currently, the following is done:

1. The `ESBMC_CONFIG` env is checked if set, if it is set to "" this explicitly disables config loading, if it's set to a path to a TOML config file it will use that. If it is set to a wrong location, ESBMC will exit with an error (if a path is specified, it needs to be correct).
2. If `ESBMC_CONFIG` is unset, then the config directory of the user profile is checked as a default location.

This is a nice and simple solution for users to understand and captures the following scenarios:

1. I can have a default config file in the user config directory.
2. If I want to override it I can use the env var to temporarily set it to disabled ("") or to another file.